ksqldb-engine/src/main/java/io/confluent/ksql/engine/KsqlEngine.java

```diff
@@ -324,6 +324,15 @@ public TransientQueryMetadata executeTransientQuery(
     }
   }
 
+  public KsqlConfig getKsqlConfig() {
```

Comment:
you should add this method to `KsqlExecutionContext`, and implement it there by getting the config from the passed in `EngineContext`. This way we can get the config out of the sandboxed engine. While you're at it, can you remove the `ksqlConfig` member from `KsqlEngine`? It should just use the config in `primaryContext`